Update coroutines-guide.md (#3012)
Kotlin/kotlinx.coroutines: coroutines-guide.md

The main coroutines guide has moved to the [docs folder](docs/topics/coroutines-guide.md) and split up into smaller documents.

> It is recommended to read this guide on [kotlinlang site](https://kotlinlang.org/docs/coroutines-guide.html) with proper HTML formatting and runnable samples.

Reviewer: ```suggestion
It is recommended to read the guide on the [kotlinlang website](https://kotlinlang.org/docs/coroutines-guide.html), with proper HTML formatting and runnable samples.
```
